### PR TITLE
communication history + adding notes

### LINKED ISSUE #477: [MicroCredit] applications notes
dependent on https://github.com/impactMarket/backend/issues/674

### PR DESCRIPTION
**Pull request by marcusats**
This PR fixes #477 

## Changes

- 

## Tests

- [ ] Tested on mobile
- [ ] Tested on dektop

This PR should be manually tested by @hugomarquesdev @obernardovieira @PauloSousapt , included on reviewers.

### CODE REVIEW

**Review by obernardovieira — COMMENTED** [review]
The fonts seems fairly small to me, specially the loan manager name and address. Is it correct?
image

**Review by obernardovieira — CHANGES_REQUESTED** [review]
When clicking the address below the loan manager name, it doesn't do anything. It should show the same options as the other addresses (open in explore & copy). I think it's a generic component

**Comment by obernardovieira** [comment]
👏  looks great :)

File: src/hooks/useAddNote.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+            setLoading(false);
+            setError(err.message);
+        }
+    };
+
+    return { addNote, error, loading };
+};
+
+export default useAddNote;
```
Comment (hugomarquesdev) [inline src/hooks/useAddNote.ts:79]: need blank line

File: src/hooks/useUserInfo.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    return {
+        error,
+        loadingUser,
+        mutate,
+        userInfo: data?.data
+    };
+};
+
+export default useUserInfo;
```
Comment (hugomarquesdev) [inline src/hooks/useUserInfo.ts:29]: need blank line

File: src/views/MicrocreditManager/ApproveRejectTab/BorrowersList.tsx [hunk 3]
```diff
@@ -508,35 +512,42 @@ const getColumns = (props: any) => {
                                 `/microcredit/form/${data?.application?.id}`
                             ),
                         title: 'Loan Application'
```
Comment (hugomarquesdev) [inline src/views/MicrocreditManager/ApproveRejectTab/BorrowersList.tsx:514]: use `loanApplication` from `extractFromView('messages')`

File: src/views/User/Microcredit/CommunicationHistory/ManagerNote.tsx [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+import config from '../../../../../config';
+import useTranslations from '../../../../libs/Prismic/hooks/useTranslations';
+import useUserInfo from 'src/hooks/useUserInfo';
+
+
+const ManagerNote = (props: {note: any }) => {
+    const { note } = props;
+    const { t } = useTranslations();
+    const { extractFromView } = usePrismicData();
+    const {} = extractFromView('messages') as any;
```
Comment (hugomarquesdev) [inline src/views/User/Microcredit/CommunicationHistory/ManagerNote.tsx:27]: is this needed?

File: src/views/User/Microcredit/CommunicationHistory/index.tsx [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+                            ))
+                        }
+                    </Box>
+                    <Box w="100%">
+                        {pageCount > 1 && (
+                            <Pagination
+                                currentPage={currentPage}
+                                handlePageClick={handlePageClick}
+                                nextIcon="arrowRight"
+                                nextLabel="Next"
```
Comment (hugomarquesdev) [inline src/views/User/Microcredit/CommunicationHistory/index.tsx:112]: use `nextLabel={t('next')}`

File: src/views/User/Microcredit/CommunicationHistory/index.tsx [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+                                pageCount={pageCount}
+                                previousIcon="arrowLeft"
+                                previousLabel="Previous"
```
Comment (hugomarquesdev) [inline src/views/User/Microcredit/CommunicationHistory/index.tsx:115]: use `previousLabel={t('previous')}`

File: src/views/User/Microcredit/index.tsx [hunk 7]
```diff
@@ -1,24 +1,126 @@
...
+                            icon="coment"
+                            onClick={()=> {
+                                openModal('addNote',
+                                    {
+                                        borrowerAdd: user?.address,
+                                    }
+                                )
+                            }}
+                        >
+                            Add Note
```
Comment (hugomarquesdev) [inline src/views/User/Microcredit/index.tsx:118]: use `addNote` from `extractFromView('microcredit')`

File: src/views/User/Microcredit/index.tsx [hunk 8]
```diff
@@ -34,8 +136,13 @@ const Microcredit = (props: { user: any }) => {
...
+                <Box mt={0.5}>
+                    <CommunicationHistory user={user}/>
+                </Box>
+            </TabPanel>
         </Tabs>
     );
 };
 
-export default Microcredit;
+export default Microcredit;
```
Comment (hugomarquesdev) [inline src/views/User/Microcredit/index.tsx:148]: need blank line

**Review by hugomarquesdev — CHANGES_REQUESTED** [review]

**Review by hugomarquesdev — APPROVED** [review]

**Pull request merged**